### PR TITLE
Added option checks for validations layers that aren't supported on all platforms

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: engine/graphics/src/vulkan/graphics_vulkan_context.cpp [hunk 1]
```diff
@@ -191,30 +207,29 @@ namespace dmGraphics
...
-            #ifdef __MACH__
-                vk_required_extensions.OffsetCapacity(2);
-                vk_required_extensions.Push(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME);
-                vk_required_extensions.Push("VK_KHR_get_physical_device_properties2");
-            #else
-                vk_required_extensions.OffsetCapacity(1);
-                vk_required_extensions.Push(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME);
-            #endif
+                AddIfSupported(extensions, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME, vk_required_extensions);
+                AddIfSupported(extensions, VK_KHR_get_physical_device_properties2_str, vk_required_extensions);
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_vulkan_context.cpp:223]: One of them weren't supported on Switch (I don't recall which one)

File: engine/graphics/src/vulkan/graphics_vulkan_context.cpp [hunk 2]
```diff
@@ -191,30 +207,29 @@ namespace dmGraphics
...
 
                 for (uint16_t i=0; i < validationLayerExtensionCount; ++i)
                 {
-                    vk_required_extensions.Push(validationLayerExtensions[i]);
+                    AddIfSupported(extensions, validationLayerExtensions[i], vk_required_extensions);
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_vulkan_context.cpp:227]: Let's varify the other validation layers as well, while we're at it.

File: engine/graphics/src/vulkan/graphics_vulkan_context.cpp (hunk 1 above)
Comment (Jhonnyg) [reply engine/graphics/src/vulkan/graphics_vulkan_context.cpp:223]: Yeah, the vulkan version on switch is probably a bit behind. Next time we need validation for switch I'll make sure to use the correct validation layers, but for now it's more important it works on osx/win32

**Review by Jhonnyg — APPROVED** [review]
👍

**Pull request merged**